realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -579,11 +579,8 @@ public long getPrimaryKey() {
                 return NO_PRIMARY_KEY; // Free table = No primary key
             }
 
-            String tableName = getName();
-            if (tableName.startsWith(TABLE_PREFIX)) {
-                tableName = tableName.substring(TABLE_PREFIX.length());
-            }
-            long rowIndex = pkTable.findFirstString(PRIMARY_KEY_CLASS_COLUMN_INDEX, tableName);
+            String className = tableNameToClassName(getName());
+            long rowIndex = pkTable.findFirstString(PRIMARY_KEY_CLASS_COLUMN_INDEX, className);
```

Comment:
The name of the all tables created from model classes have "class_" prefix and the model name in pkTable does not have that prefix.

I don't think we can find correct `rowIndex` if we omit the `tableNameToClassName()` here.